Kubernetes/windows/install.ps1

```diff
@@ -0,0 +1,84 @@
+    [parameter(Mandatory = $false)] $LogDir = "C:\k"
+)
+
+function SetupDirectories($LogDir)
+{
+    Write-Host "Creating Kubernetes directories"
+    md $BaseDir -ErrorAction Ignore
+    md $LogDir -ErrorAction Ignore
+    md c:\flannel -ErrorAction Ignore
+    md $BaseDir\cni\config -ErrorAction Ignore
+    md C:\etc\kube-flannel -ErrorAction Ignore
+}
+
+function CopyFiles()
+{
+    Write-Host "Copying Flannel setup files"
+    cp $BaseDir\flanneld.exe c:\flannel\flanneld.exe
+    cp $BaseDir\net-conf.json C:\etc\kube-flannel\net-conf.json
+}
+
+function DownloadFlannelBinaries()
+{
+    Write-Host "Downloading Flannel binaries"
+    DownloadFile -Url  "https://github.com/coreos/flannel/releases/download/v0.11.0/flanneld.exe" -Destination $BaseDir\flanneld.exe 
+}
+
+function DownloadCniBinaries($NetworkMode)
+{
+    Write-Host "Downloading CNI binaries"
+    DownloadFile -Url  "https://github.com/Microsoft/SDN/raw/master/Kubernetes/flannel/$NetworkMode/net-conf.json" -Destination $BaseDir\net-conf.json
+    DownloadFile -Url "https://github.com/Microsoft/SDN/raw/master/Kubernetes/flannel/$NetworkMode/cni/config/cni.conf" -Destination $BaseDir\cni\config\cni.conf
+    DownloadFile -Url  "https://github.com/Microsoft/SDN/raw/master/Kubernetes/flannel/l2bridge/cni/flannel.exe" -Destination $BaseDir\cni\flannel.exe
+    DownloadFile -Url  "https://github.com/Microsoft/SDN/raw/master/Kubernetes/flannel/l2bridge/cni/host-local.exe" -Destination $BaseDir\cni\host-local.exe
+
+    if ($NetworkMode -eq "l2bridge")
+    {
+        DownloadFile -Url  "https://github.com/Microsoft/SDN/raw/master/Kubernetes/flannel/l2bridge/cni/win-bridge.exe" -Destination $BaseDir\cni\win-bridge.exe
+    }
+    elseif ($NetworkMode -eq "overlay"){
+        DownloadFile -Url  "https://github.com/Microsoft/SDN/raw/master/Kubernetes/flannel/overlay/cni/win-overlay.exe" -Destination $BaseDir\cni\win-overlay.exe
+    }
+}
+
+function DownloadWindowsKubernetesScripts($NetworkMode)
+{
+    Write-Host "Downloading Windows Kubernetes scripts"
+    DownloadFile -Url  https://github.com/Microsoft/SDN/raw/master/Kubernetes/windows/hns.psm1 -Destination $BaseDir\hns.psm1
+    DownloadFile -Url  https://github.com/Microsoft/SDN/raw/master/Kubernetes/windows/InstallImages.ps1 -Destination $BaseDir\InstallImages.ps1
+    DownloadFile -Url  https://github.com/Microsoft/SDN/raw/master/Kubernetes/windows/Dockerfile -Destination $BaseDir\Dockerfile
+    DownloadFile -Url  https://github.com/Microsoft/SDN/raw/master/Kubernetes/flannel/stop.ps1 -Destination $BaseDir\stop.ps1
+    DownloadFile -Url  https://github.com/Microsoft/SDN/raw/master/Kubernetes/flannel/$NetworkMode/start-kubelet.ps1 -Destination $BaseDir\start-Kubelet.ps1
+    DownloadFile -Url  https://github.com/Microsoft/SDN/raw/master/Kubernetes/flannel/$NetworkMode/start-kubeproxy.ps1 -Destination $BaseDir\start-Kubeproxy.ps1
+}
+

```

Comment:
As part of a follow up PR, Can we also add a method to download the kubernetes binary? It would reduce one more manual step

Reply:
Also the current scripts are updated which uses features from latest kubernetes.
Am not sure if we have to maintain scripts for older kubenetes version. eg. kube proxy changes

Reply:
True, I removed the WinDSR feature gate. For overlay, this is a new feature we are advertising with v1.14 so it should be OK to keep it.